README.md
WordPlate
=========

![wordplate](https://cloud.githubusercontent.com/assets/499192/9404207/ba83bdc4-47ed-11e5-9e30-269c0c343b55.png)

A [WordPress](https://github.com/WordPress/WordPress) boilerplate. This project is trying to simplify the way we're setting up a new WordPress project. [Don't repeat yourself](http://en.wikipedia.org/wiki/Don't_repeat_yourself).

[![Build Status](https://img.shields.io/travis/wordplate/framework/master.svg?style=flat)](https://travis-ci.org/wordplate/framework)
[![StyleCI](https://styleci.io/repos/13329845/shield?style=flat)](https://styleci.io/repos/13329845)
[![Latest Version](https://img.shields.io/github/release/wordplate/wordplate.svg?style=flat)](https://github.com/wordplate/wordplate/releases)
[![License](https://img.shields.io/packagist/l/wordplate/wordplate.svg?style=flat)](https://packagist.org/packages/wordplate/wordplate)

## Installation

Install WordPlate by issuing the Composer `create-project` command in your terminal:

```bash
composer create-project wordplate/wordplate
```

Read more about setting up WordPlate on our [installation documentation](https://github.com/wordplate/wordplate/wiki/installation) page.

## Documentation

WordPlate features an extensive [wiki documentation](https://github.com/wordplate/wordplate/wiki) to help you getting started. [Take me to the docs!](https://github.com/wordplate/wordplate/wiki)

## Contributing

Please review our [contribution guidelines](https://github.com/wordplate/framework/blob/master/CONTRIBUTING.md) before submitting a pull request.

## License

WordPlate is licensed under [The MIT License (MIT)](LICENSE).


CHANGELOG.md
# CHANGELOG

## 2.7.0 (released 2015-08-20)

- Updated to WordPress 4.3
- Removed unused server page
- Removed the ability to hide updates
- Code cleanup

## 2.6.0 (released 2015-08-18)

- Added [Elixir](http://laravel.com/docs/elixir) Support (#117)

## 2.5.4 (released 2015-07-01)

- Sanitize File Name on Save (#113)

## 2.5.3 (released 2015-06-02)

- Fix Mail From and Name Bug (#112)

## 2.5.2 (released 2015-05-28)

- Added Charset and Language Attributes to Header (#111)
- Added wp_nav_menu to Header (#108)
- Fixed wp_title Category Bug (#107)

## 2.5.1 (released 2015-05-21)

- Remove Clear Command (#105)

## 2.5.0 (released 2015-05-21)

- Fix Permalink Update Method Bug (#104)
- Move Docs to Website (#102)
- Add Test Coverage (#103)
- Add Mail Configuration (#100)

## 2.4.0 (released 2015-05-10)

- Add post revisions to config (#98)
- Add IMAGE_EDIT_OVERWRITE (#94)
- Disable WordPress core updates (#97)
- Disable plugin and theme update and installation (#95)
- Make Whoops to handle all errors (#93)

## 2.3.0 (released 2015-05-03)

- Add plate console commander (#92)
- Add plate salts generator command (#81)
- Add new exception handler (#91)
- Add WP_HOME and WP_SITEURL (#89)
- Update Soil plugin to latest version (#90)
- Rename wp-content directory to public (#88)

## 2.2.0 (released 2015-04-26)

- Add theme color meta tag (#84)
- Add title tag theme support (#83)
- Add Windows support to post-create-project-cmd (#82)
- Add HTML5 tags theme support (#77)
- Update configuration files structure (#79)
- Update WordPress to version 4.2 (#78)
- Update header HTML (#85)
- Update @roots Soil plugin version (#86)
- Remove ACF simple editor toolbar (#87)

## 2.1.3 (released 2015-04-22)

- Fix template bug for pages (#80)

## 2.1.2 (released 2015-04-21)

- Fix infinite pages issue (#76)

## 2.1.1 (released 2015-04-21)

- Deregister jQuery by default (#73)
- Fix login image URL bug (#74)
- Fix infinite pages if the title key isn't available (#75)

## 2.1.0 (released 2015-04-19)

- Added default screenshot and favicon (#71)
- Added new directory structure (#70)
- Added default pages (#67)
- Added Google Analytics to configuration files (#68)
- Added plugin activation on theme switch (#66)

## 2.0.0 (released 2015-04-16)

- Rename the project to WordPlate
- Moved project to an organization
- Moved library to framework repository
- Added installer CLI tool
- Added theme activation on theme setup (#56)
- Added new installation guide
- Added changelog
- Update @roots soil plugin to the latest version

## 1.3.0 (released 2015-04-10)

- Added TinyPGN plugin (#57)
- Added @laravel's helper methods (#63)
- Added rename the home URL on theme setup (#64)

## 1.2.2 (released 2015-04-10)

- Added Whoops php errors for cool kids (#62)

## 1.2.1 (released 2015-04-10)

- Added configuration to disbale/enable updates

## 1.2.0 (released 2015-03-31)

- Added config helper method (#61)

## 1.1.1 (released 2015-03-27)

- Added WP Sweep plugin by @lesterchan (#59)

## 1.1.0 (released 2015-03-20)

- Added Dotenv environment package by @vlucas (#44)
- Added @roots Soil plugin (#49)
- Added Google Analytics (#52)

## 1.0.0 (released 2015-03-20)

- Move Boilerplate to Composer
- Added Server Settings page
- Added Custom Post Type class
- Added a boilerplate theme filled with actions and filters
- Added WordPress Packagist, WordPress plugins as packages
- Added plugins such as Advanced Custom Fields and WordPress Importer


lib/config.php
<?php

return array(

	/*
	|--------------------------------------------------------------------------
	| Footer Text
	|--------------------------------------------------------------------------
	|
	| Replace the default, Thank you for creating with WordPress, footer text.
	| Specially good when building themes for clients. Promoting yourself.
	|
	*/

	'footer_text' => 'Thank you for creating with <a href="http://vinkla.com">Vincent Klaiber</a>.',

	/*
	|--------------------------------------------------------------------------
	| Login Logo
	|--------------------------------------------------------------------------
	|
	| Replace the default login logo with clients logo or name will give your
	| custom built theme an extra touch.
	|
	| Define the width of your custom login logo in pixels.
	|
	*/

	'login_image_path' => get_template_directory_uri().'/images/admin-login-logo.png',
	'login_image_width' =>  80,

	/*
	|--------------------------------------------------------------------------
	| Login Error Message
	|--------------------------------------------------------------------------
	|
	| It is good practice to replace the login error message for security
	| reasons. We don't wanna give unwanted guests any feedback except that
	| he/she can't login without the right credentials.
	|
	*/

	'login_error_message' => 'Whoops! Looks like you missed something there. Have another go.',

	/*
	|--------------------------------------------------------------------------
	| JPEG Quality
	|--------------------------------------------------------------------------
	|
	| WordPress does automatically compress JPEG images at 90% quality. Setting
	| it to 100 would mean that WordPress would compress the image at its
	| highest quality.
	|
	*/

	'jpeg_quality' => 100,

	/*
	|--------------------------------------------------------------------------
	| Block External HTTP Requests
	|--------------------------------------------------------------------------
	|
	| On each page load in the administrator area, WordPress sends HTTP
	| requests to check for new updates for themes and plugins. To block
	| external HTTP requests right in WordPress's core itself set this to true.
	|
	*/

	'http_block_external' => true,

	/*
	|--------------------------------------------------------------------------
	| Disallow File Edit
	|--------------------------------------------------------------------------
	|
	| Occasionally you may wish to disable the plugin or theme editor to
	| prevent overzealous users from being able to edit sensitive files and
	| potentially crash the site.
	|
	*/

	'disallow_file_edit' => true,

	/*
	|--------------------------------------------------------------------------
	| TinyMCE editor
	|--------------------------------------------------------------------------
	|
	| All tool bar options in WordPress default WYSIWYG editor that doesn't
	| have any custom styling should be removed.
	|
	| Full list of formats http://www.tinymce.com/wiki.php/Controls
	|
	*/

	'tinymce_blockformats' => array('p', 'h2', 'h3'),
	'tinymce_disabled' => array(
		'strikethrough',
		'underline',
		'forecolor',
		'justifyfull'
	),

	/*
	|--------------------------------------------------------------------------
	| Remove Menu Items
	|--------------------------------------------------------------------------
	|
	| Giving clients and users access to everything within the administrator
	| dashboard is bad practice. Try to give them access to what they actually
	| need and will use.
	|
	| Specified below which menu items should be deleted for which users.
	|
	| Available items: appearance, comments, dashboard, media, plugins, pages,
	| posts, settings, tools, users
	|
	*/

	'remove_menu_items' => array(
		// removed for user without administrator capabilities
		'default' => array(
			'appearance',
			'comments',
			'dashboard',
			'media',
			'plugins',
			'settings',
			'tools',
			'users'
		),
		// removed for user with administrator capabilities
		'administrator' => array(
			'comments',
			'dashboard',
			'media'
		)
	),

	/*
	|--------------------------------------------------------------------------
	| Remove Menu Bar Links
	|--------------------------------------------------------------------------
	|
	| The menu bar in WordPress administrator dashboard looks cluttered.
	| Remove unnecessary links, users should only see what the actually need.
	|
	*/

	'remove_menu_bar_links' => array(
		'comments',
		'wp-logo',
		'edit',
		'appearance',
		'view',
		'new-content',
		'updates',
		'search'
	),

	/*
	|--------------------------------------------------------------------------
	| Remove Dashboard Widgets
	|--------------------------------------------------------------------------
	|
	| Remove dashboard widgets to cleanup the administrator dashboard. Most
	| clients and user will never even bother using them. Instead, replace the
	| widgets with something more useful like Yahoo weather reports ;).
	|
	*/

	'remove_dashboard_widgets' => array(
		'side' => array(
			'dashboard_primary',
			'dashboard_secondary',
			'dashboard_quick_press',
			'dashboard_recent_drafts'
		),
		'normal' => array(
			'dashboard_plugins',
			'dashboard_recent_comments',
			'dashboard_incoming_links',
			'dashboard_right_now',
			'dashboard_activity'
		)
	),

	/*
	|--------------------------------------------------------------------------
	| Remove Meta Boxes
	|--------------------------------------------------------------------------
	|
	| Remove unnecessary custom fields and meta boxes to cleanup the edit posts
	| view. Lets be honest, how often do you even use some of these boxes?
	|
	*/

	'meta_boxes' => array(
		'link' => array(
			'linktargetdiv',
			'linkxfndiv',
			'linkadvanceddiv',
		),
		'post' => array(
			'postexcerpt',
			'trackbacksdiv',
			'postcustom',
			'commentstatusdiv',
			'commentsdiv',
			'revisionsdiv',
			'authordiv',
			'sqpt-meta-tags',
			'slugdiv',
			//'categorydiv',
			//'tagsdiv-post_tag'
		)
	),

	/*
	|--------------------------------------------------------------------------
	| Disable Default Widgets
	|--------------------------------------------------------------------------
	|
	| Hide and disable default widgets within WordPress core. Add widgets to
	| the list to leave them out of WordPress admin.
	|
	*/

	'widgets' => [
		'WP_Widget_Pages',
		'WP_Widget_Calendar',
		'WP_Widget_Archives',
		'WP_Widget_Meta',
		'WP_Widget_Search',
		'WP_Widget_Categories',
		'WP_Widget_Recent_Posts',
		'WP_Widget_Recent_Comments',
		'WP_Widget_RSS',
		'WP_Widget_Tag_Cloud'
	],

	/*
	|--------------------------------------------------------------------------
	| Hide Panel Tabs
	|--------------------------------------------------------------------------
	|
	| Remove and hide screen options and help panel tab. This will make the
	| administrator interface more cleaned up for users.
	|
	*/

	'panel_tabs' => array(
		'help' => false,
		'screen_options' => false
	)
);
